coral-component-masonry/src/scripts/Masonry.js

```diff
@@ -517,6 +651,10 @@ class Masonry extends BaseComponent(HTMLElement) {
       }
     }
     
+    // Update items, so that column indexes are correctly set
+    this._updateAriaRoleForItems(this.ariaGrid);
+    this._updateAriaColumnCountForParent(this.ariaGrid);
```

Comment:
If _doLayout is called while the masonry is initialised, then the methods will be called with ariaGrid.OFF, resulting to remove attributes which should not happen. I suggest to check when _doLayout is called.

Reply:
_doLayout is being called on resizing and on addition of new items. 
I have added tests, to verify if attributes are added or removed in case ariagrid is enabled as through setting attribute in HTML or dynamically.